data/trie/evictionWaitingList/evictionWaitingList.go

```diff
@@ -0,0 +1,91 @@
+package evictionWaitingList
+
+import (
+	"github.com/ElrondNetwork/elrond-go/data/trie"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+// evictionWaitingList is a structure that caches keys that need to be removed from a certain database.
+// If the cache is full, the keys will be stored in the underlying database. Writing at the same key in
+// cacher and db will overwrite the previous values
+type evictionWaitingList struct {
+	cache       map[string][][]byte
+	cacheSize   int
+	db          storage.Persister
+	marshalizer marshal.Marshalizer
+}
+
+// NewEvictionWaitingList creates a new instance of evictionWaitingList
+func NewEvictionWaitingList(size int, db storage.Persister, marshalizer marshal.Marshalizer) (*evictionWaitingList, error) {
+	if size < 1 {
+		return nil, trie.ErrInvalidCacheSize
+	}
+	if db == nil || db.IsInterfaceNil() {
+		return nil, trie.ErrNilDatabase
+	}
+	if marshalizer == nil || marshalizer.IsInterfaceNil() {
+		return nil, trie.ErrNilMarshalizer
+	}
+
+	return &evictionWaitingList{
+		cache:       make(map[string][][]byte),
+		cacheSize:   size,
+		db:          db,
+		marshalizer: marshalizer,
+	}, nil
+}
+
+// Put stores the given hashes in the eviction waiting list, in the position given by the root hash
+func (ec *evictionWaitingList) Put(rootHash []byte, hashes [][]byte) error {
```

Comment:
maybe change the receiver's name to _ewl_ in all methods from this file

Reply:
done.